projectblacklight/spotlight
# Replacing header nav partial with header component

**Context**
Relates to having tests pass with Blacklight 8.3 and using a Spotlight HeaderComponent instead of shared/_header_navbar.  The Spotlight header is almost exactly the same as the default Blacklight header except it does not include the search bar (which is elsewhere on the page). 
Refer to https://github.com/projectblacklight/spotlight/issues/3047 and this comment which includes a checklist of tests to fix: https://github.com/projectblacklight/spotlight/issues/3047#issuecomment-2359456286

For 8.3, the following tests are failing:
 - rspec ./spec/views/shared/_header_navbar.html.erb_spec.rb:8 # shared/_header_navbar has nav links
 - rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:46 # spotlight/pages/show when rendering with layout does not double-escape HTML entities in the HTML title
 - rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:50 # spotlight/pages/show when rendering with layout includes analytics reporting

**What this pull request does**
- Removes the header_navbar partial.
- Creates a Spotlight HeaderComponent whose partial displays on the top bar. 
- Adds the Spotlight Header Component to the configuration as the header component.
- Adds blacklight configuration information for the show html rspec to pass. 

**How to test**
These tests should still pass in Blacklight 7.

How to test with Blacklight 8: Generate test app (.internal_test_app). Update the internal test app's Gemfile to use Blacklight 8.3. Run bundle install for the internal test app. Then run rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:46 and rspec ./spec/views/spotlight/pages/show.html.erb_spec.rb:50.
